In the fitted model equation for lm, added a hat to the response variable and also hide the epsilon term.

thomas-fung opened the pull request:
``` r
library(equatiomatic)
ctl <- c(4.17,5.58,5.18,6.11,4.50,4.61,5.17,4.53,5.33,5.14)
trt <- c(4.81,4.17,4.41,3.59,5.87,3.83,6.03,4.89,4.32,4.69)
group <- gl(2, 10, 20, labels = c("Ctl","Trt"))
weight <- c(ctl, trt)
lm.D9 <- lm(weight ~ group)
extract_eq(lm.D9, use_coefs = FALSE)
```

\[
\operatorname{weight} = \alpha + \beta_{1}(\operatorname{group}_{\operatorname{Trt}}) + \epsilon
\]

``` r
extract_eq(lm.D9, use_coefs = TRUE)
```

\[
\operatorname{\widehat{weight}} = 5.03 - 0.37(\operatorname{group}_{\operatorname{Trt}})
\]

<sup>Created on 2020-08-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

datalorax commented:
Thanks @thomas-fung! I think this makes a lot of sense! I'm in favor of merging this in. Thoughts @andrewheiss?

datalorax commented:
Also, @thomas-fung, if you wouldn't mind adding one additional commit to modify the DESCRIPTION file to (a) add yourself as a contributor and (b) modify the NEWS.md to reflect this change that would be excellent!

andrewheiss commented:
I like it!

datalorax commented:
Okay, awesome! @thomas-fung if you can make that one additional commit with the pieces mentioned above we'll get this merged in. I'm taking off for the weekend here in a bit and won't have internet but Andrew can merge it in if you don't get to it before I leave.

thomas-fung commented:
Thanks @datalorax and @andrewheiss! All done.

thomas-fung commented:
Oops. These changes should probably go into a different (or may be even multiple) pull request. 

1. Fixed the problem that error terms keep showing up in the model equation. Adding class(model) to class(lhs) doesn't quite work so create_eq (the default one) is used all the time. I moved the model argument to the first position and that seems to be able to fix the problem. 
``` r
library(equatiomatic)
set.seed(1234)
df <- data.frame(outcome = sample(0:1, 300, replace = TRUE),
                 categorical = rep(letters[1:3], 100),
                 continuous_1 = rnorm(300, 100, 1),
                 continuous_2 = rnorm(300, 50, 5))

model_logit <- glm(outcome ~ ., data = df,
                   family = binomial(link = "logit"))

extract_eq(model_logit)
```

\[
\log\left[ \frac { P( \operatorname{outcome} = \operatorname{1} ) }{ 1 - P( \operatorname{outcome} = \operatorname{1} ) } \right] = \alpha + \beta_{1}(\operatorname{categorical}_{\operatorname{b}}) + \beta_{2}(\operatorname{categorical}_{\operatorname{c}}) + \beta_{3}(\operatorname{continuous\_1}) + \beta_{4}(\operatorname{continuous\_2})
\]

<sup>Created on 2020-08-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
2. The distribution assumption for the logistic regression is technically a Bernoulli instead of a binomial as binomial implies that the trails are all iid. Do you think the following notation would work better? 
``` r
library(equatiomatic)
set.seed(1234)
df <- data.frame(outcome = sample(0:1, 300, replace = TRUE),
                 categorical = rep(letters[1:3], 100),
                 continuous_1 = rnorm(300, 100, 1),
                 continuous_2 = rnorm(300, 50, 5))

model_logit <- glm(outcome ~ ., data = df,
                   family = binomial(link = "logit"))

extract_eq(model_logit, show_distribution = TRUE)
```

\[
\begin{aligned}
\operatorname{outcome} &\sim Bernoulli\left(\operatorname{prob}_{\operatorname{outcome} = \operatorname{1}}= \hat{P}\right) \\
 \log\left[ \frac { \hat{P} }{ 1 - \hat{P} } \right] 
 &= \alpha + \beta_{1}(\operatorname{categorical}_{\operatorname{b}}) + \beta_{2}(\operatorname{categorical}_{\operatorname{c}}) + \beta_{3}(\operatorname{continuous\_1}) + \beta_{4}(\operatorname{continuous\_2}) + \epsilon
\end{aligned}
\]

<sup>Created on 2020-08-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup> 
3. Now support basic glm eqn display for all distributions as well as support for the offset term. I removed the P() notation from link_formula.  

``` r
library(equatiomatic)
set.seed(1234)
df <- data.frame(outcome = sample(0:15, 300, replace = TRUE),
                 categorical = rep(letters[1:3], 100),
                 continuous_1 = rnorm(300, 100, 1),
                 continuous_2 = rnorm(300, 50, 5),
                 offset = rep(1, 300))

model_log <- glm(outcome ~ categorical + 
                   continuous_1 + offset(offset), data = df,
                   family = poisson(link = "log"))

extract_eq(model_log)
```

\[
\log ({ E( \operatorname{outcome} ) })  = \alpha + \beta_{1}(\operatorname{categorical}_{\operatorname{b}}) + \beta_{2}(\operatorname{categorical}_{\operatorname{c}}) + \beta_{3}(\operatorname{continuous\_1}) + \operatorname{offset(offset)}
\]

<sup>Created on 2020-08-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

4. Fixed the issue that log is also a pattern match to "logistic" and "logit" 
5. Fixed the issue that "link" should be saved in "info" instead of "family" for `clm` objects.

Also notice that in the "Distribution-based equations work" part of text-glm.R, there are whitespaces at the end of lines 70 and 76 so that it matches the function output. As one of the project options in RStudio is to strip trailing horizontal whitespace when saving which means these things could be changed accidentally and failed the test.

datalorax commented:
Thanks @thomas-fung! I think this all looks great. I'm okay with keeping these all in the same PR but it would be nice to have separate PRs in the future. I've given it all a quick scan but need to dig into it a bit more. I feel fairly confident it's good to go though and will likely merge later today.

datalorax commented:
So one thing I'm just noticing - it looks like the subscripts for the betas are being dropped after the first one. In other words, it should be showing up as `\beta_{2}` instead of `\beta{2}`. Not sure why this is happening.

datalorax commented on R/create_eq.R:
```diff
@@ -25,35 +26,40 @@ create_eq.default <- function(lhs, rhs, ital_vars, use_coefs, coef_digits,
   error_row$term <- "error"
   error_row$final_terms <- "\\epsilon"
-  rhs <- rbind(rhs, error_row)
+  rhs <- rbind(rhs, error_row)}
```
Super minor but can you bump this close curly brace to the next line?

thomas-fung replied:
Fixed.

datalorax commented on R/create_eq.R:
```diff
@@ -25,35 +26,40 @@ create_eq.default <- function(lhs, rhs, ital_vars, use_coefs, coef_digits,
   }
+  if (!is.null(model$offset)){
+    rhs <- rbind(rhs, c(rep(NA, (dim(rhs)[2]-1)),
+                        add_tex_ital(utils::tail(names(attr(model$terms, "dataClasses")),1), ital_vars)))
```
In the equatiomatic-package.R file, can you add `#' @importFrom utils tail`, then run `devtools::document`, and then delete the namespacing here?

thomas-fung replied:
Fixed.

datalorax commented on R/extract_lhs.R:
```diff
@@ -35,10 +35,14 @@ extract_lhs2 <- function(model, ...) {
 
   lhs_escaped <- escape_tex(lhs)
+  if (use_coefs){
+  lhs_escaped <- add_hat(lhs)}
   add_tex_ital_v(lhs_escaped, ital_vars)
```
Again super minor, but can you indent these two lines two spaces?

thomas-fung replied:
Fixed

datalorax commented on R/extract_lhs.R:
```diff
@@ -110,32 +147,27 @@ extract_lhs2.glm <- function(model, ital_vars, ...) {
-    )
-  }
+  lhs <- add_tex_ital_v(outcome_escaped, ital_vars)
+  p <- paste0("\\operatorname{prob}",
```
Is `prob` ideal here? I guess I'm wondering why not just `P`.

thomas-fung replied:
It can definitely be just `P`. Fixed.

datalorax commented on R/extract_lhs.R:
```diff
@@ -209,6 +241,9 @@ modify_lhs_for_link.glm <- function(model, lhs) {
     model$family$link <- "identity"
   }
   matched_row_bool <- grepl(model$family$link, link_function_df$link_name)
+  if (sum(matched_row_bool)>1){
```
Again really minor but, please add spaces around `>`.

thomas-fung replied:
Fixed.

datalorax commented:
Hi @thomas-fung, just wanted to check in to see if you're willing to implement these changes so we can get it merged in?

thomas-fung commented:
Hi @datalorax, I am happy to make the changes. Will take a look at them today. Thanks!

datalorax commented:
Thanks! Merging in now.
